elements/pl-code/pl-code.py

```diff
@@ -202,19 +232,18 @@ def render(element_html, data):
         #
         # which technically starts with a newline, but we probably
         # don't want a blank line at the start of the code block.
-        code = pl.inner_html(element)
-        if len(code) > 1 and code[0] == "\r" and code[1] == "\n":
-            code = code[2:]
-        elif len(code) > 0 and (code[0] == "\n" or code[0] == "\r"):
-            code = code[1:]
+        code = pl.inner_html(element).removeprefix("\r").removeprefix("\n")  # type: ignore
 
-    if specify_language:
-        lexer = get_lexer_by_name(language)
-    else:
-        lexer = NoHighlightingLexer()
+    lexer = NoHighlightingLexer() if language is None else get_lexer_by_name(language)
+
+    pygments_style = get_style_by_name(style)
+
+    class CustomStyleWithAnsiColors(pygments_style):  # type: ignore
```

Comment:
I opened https://github.com/python/typeshed/pull/9803 to remove the need for `# type: ignore` here.

Reply:
I think this is actually unavoidable, Python typechecking doesn't like dynamic class creation, see https://github.com/microsoft/pyright/issues/2058

Reply:
I tested this change locally and it worked; is there something else that you're thinking will be broken?

Reply:
Oh interesting, I guess pyright got updated to work in these situations or something.